### ISSUE TITLE
bug: Ion Select is behind Ion Modal

### ISSUE THREAD

**Issue by BerkeAras** [ionic-team/ionic-framework#21181]
# Bug Report

**Ionic version:**

[x] **5.x**

**Current behavior:**

ion-select is behind ion-modal.

**Expected behavior:**

ion-select should be over ion-modal.

**Steps to reproduce:**

Open a ion-modal on iOS. In this modal open ion-select.

**Comment by JEricaM** [comment]
I have this problem too. How can be solved?

**Comment by gbj** [comment]
I'm experiencing this same behavior in one case and not another, which may help point toward a solution. I am building a Stencil component that is imported into an Angular app. When I use the component in the Stencil dev server, the select alert appears over the modal correctly

<img width="620" alt="image" src="https://user-images.githubusercontent.com/286622/84498212-2ed83700-ac7e-11ea-9901-e1dbc7c67ca6.png">

When I use the component within an Angular app, the select is behind the modal and does not appear. (No image necessary for this one...)

In the Stencil version, `ion-modal` and `ion-alert` are siblings

<img width="466" alt="image" src="https://user-images.githubusercontent.com/286622/84498286-5e873f00-ac7e-11ea-8b5a-3030a278f518.png">

In the Angular version, `ion-modal` is appended to `body`, but `ion-alert` for the select is nested within `app-root` (`venite-app` in my case) > `ion-app`.

<img width="485" alt="image" src="https://user-images.githubusercontent.com/286622/84498441-aa39e880-ac7e-11ea-9100-b998243ec216.png">

**Comment by gbj** [comment]
I see the advice given by @mhartington [here](https://forum.ionicframework.com/t/on-select-is-hidden-behind-ion-modal/189703/3) is to import and use `modalController` instead of manually appending a modal, which is the method in the docs. See issue in the `ionic-docs` repo [here](https://github.com/ionic-team/ionic-docs/issues/1428)

```
import { Component, h } from '@stencil/core';
import { modalController } from '@ionic/core';
@Component({
  tag: 'app-home',
  styleUrl: 'app-home.css',
})
export class AppHome {
  async openModal() {
    const modal = await modalController.create({
      component: 'app-modal',
    });
    modal.onWillDismiss().then((x) => {
      console.log(x);
    });

    await modal.present();
  }
```
> TL;DR, don’t create modals manually. Use the built in controllers to do this.

**Comment by liamdebeasi** [comment]
Hi everyone,

I am not able to reproduce this issue using the latest version of Ionic 6, so it appears to have been fixed. I am going to close this, but let me know if you are still running into issues. Thanks!
